publish_courses/tests/test_api.py

```diff
@@ -1,11 +1,111 @@
+
+
+class ApiTestCase(TestCase):
+
+    def setUp(self):
+        super(ApiTestCase, self).setUp()
+
+        self.post_data = {
+            'account': 'sis_account_id:school:abc',
+            'term': 'sis_term_id:2015-1'
+        }
+        self.lti_data = {
+            'custom_canvas_user_login_id': '1',
+            'custom_canvas_account_sis_id': 'school:abc',
+        }
+        self.get_data = {
+            'account': 'sis_account_id:school:abc',
+            'term': 'sis_term_id:2015-1'
+        }
+        self.user = Mock(is_authenticated=Mock(return_value=True))
+        self.post_url = '/publish_courses/api/jobs'
+        self.get_url = '/publish_courses/api/show_summary'
+
+    def _prep_request_and_post(self, data):
+        request_kwargs = {
+            'content_type': 'application/json',
+            'data': json.dumps(self.post_data),
+        }
+
+        self.request = RequestFactory().post(self.post_url, **request_kwargs)
+        # these values don't matter; the keys need to be present
+        # for the authorization decorator patches to work
+        self.request.LTI = self.lti_data
+        self.request.data = data
+        self.request.user = self.user
+        bulk_publish = api.BulkPublishListCreate()
+        bulk_publish.request = self.request
+        return bulk_publish.create(self.request)
+
+    def _prep_request_and_get(self, query_params=None, data=None):
+        self.request = RequestFactory().get(self.get_url, data=data)
+        self.request.LTI = self.lti_data
+        self.request.user = self.user
+        self.request.query_params = query_params
+
+        summary_list = api.SummaryList()
+        summary_list.request = self.request
+        # return api.SummaryList().list(self.request)
+        return summary_list.list(self.request)
+
+    # Tests for BulkPublishListCreate:
+    def test_bulk_publish_list_create_success(self):
+        """
+        returns data as expected on success
+        """
+        data = {
+            'account': 'abc',
+            'term': '2015-1'
+        }
+        response = self._prep_request_and_post(data=data)
+        # validate data and status code on success
+        self.assertEqual(response.status_code, 201)
+        self.assertEqual(response.data['state'], 'queued')
+        self.assertNotEquals(response.data['date_created'], None)
+
+    def test_bulk_publish_list_create_invalid_account(self):
+        """
+        invalid account raises permission denied
+        """
+        data = {
+            'account': 'someaccount',
+            'term': '2015-1'
+        }
+        with self.assertRaises(PermissionDenied):
+            self._prep_request_and_post(data)
+
+    def test_bulk_publish_list_create_missing_data(self):
+        """
+        Invalid/missing data raises Exception
+        """
+        data = {}
+        with self.assertRaises(Exception):
+            self._prep_request_and_post(data)
+
+    # todo
+    # exception raises 500
```

Comment:
we could duplicate `test_bulk_publish_list_create_missing_data()` with a live server, but I think it's fine to trust that DRF will return a 500 if we raise a standard `Exception`, as it's supposed to.